rgw/v2/tests/s3cmd/test_rate_limit.py

```diff
@@ -0,0 +1,313 @@
+            f"put file1 s3://{bucket_name}/files/file$i ;done;"
+        )
+        # rc = utils.exec_shell_cmd(cmd)
+        rc = subprocess.Popen(
+            [cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+    # sleep a minute to reset the ops limit
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # test the read and write data limit
+    try:
+        range_val = "{1..3}"
+        cmd = (
+            f"for i in {range_val}; do /home/cephuser/venv/bin/s3cmd ls "
+            f"s3://{bucket_name}/files/file1 ;done;"
+        )
+        # rc = utils.exec_shell_cmd(cmd)
+        rc = subprocess.Popen(
+            [cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+    try:
+        range_val = "{1..3}"
+        cmd = (
+            f"for i in {range_val}; do /home/cephuser/venv/bin/s3cmd "
+            f"put file1 s3://{bucket_name}/files/file$i ;done;"
+        )
+        # rc = utils.exec_shell_cmd(cmd)
+        rc = subprocess.Popen(
+            [cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # Set the rate limits for the user and enable them
+    utils.exec_shell_cmd(
+        f"radosgw-admin ratelimit disable --ratelimit-scope=bucket --bucket={bucket_name}"
+    )
+    limset = utils.exec_shell_cmd(
+        f"radosgw-admin ratelimit set --ratelimit-scope=user --uid={user_name}"
+        + " --max-read-ops=2 --max-read-bytes=4096"
+        + " --max-write-bytes=4096 --max-write-ops=2"
+    )
+    log.info(f"Rate limits set on user {user_name}")
+    limenable = utils.exec_shell_cmd(
+        f"radosgw-admin ratelimit enable --ratelimit-scope=user --uid={user_name}"
+    )
+    limget = utils.exec_shell_cmd(
+        f"radosgw-admin ratelimit get --ratelimit-scope=user --uid={user_name}"
+    )
+    log.info(f"Rate limits enabled on bucket : {limget} ")
+
+    # test the read and write ops limit
+    bucket_name2 = utils.gen_bucket_name_from_userid(user_name, rand_no=1)
```

Comment:
Script is currently supporting only with 2 bucket. Can we make it in for loop so that we can scale bucket number